env0/validators.go

```diff
@@ -129,3 +117,29 @@ func ValidateTtl(i interface{}, path cty.Path) diag.Diagnostics {
 
 	return nil
 }
+
+func NewRoleValidator(supportedBuiltInRoles []string) schema.SchemaValidateDiagFunc {
+	return func(i interface{}, p cty.Path) diag.Diagnostics {
+		role := i.(string)
+
+		if role == "" {
+			return diag.Errorf("may not be empty")
+		}
+
+		if !client.IsBuiltinRole(role) {
+			// Custom role.
```

Comment:
even better to create `IsCustomRole` for explicit readability.
inside the common file

Reply:
Sure.